### PR TITLE
refactor newsletter signup service

### PR DESCRIPTION
**Pull request by willblake01**
More elegant and streamlined in less lines of code.

-refactor newsletter signup controller to use a struct and double splat
-reduce error handling to a rescue block
-rename api call method
-refactor .perform method test
-add exists test to verify class of object
-return service response object from .perform
[finishes #175460247]

### CODE REVIEW

**Review by runbeyond — APPROVED** [review]

File: config/routes.rb [hunk 1]
```diff
@@ -29,13 +29,14 @@
...
-  resources :sponsors, only: [:index]
-  resources :series, only: [:show]
-  resources :team_members, path: "team", only: [:index, :show]
 
   put "newsletter_signup", to: "newsletter_signups#create"
   match "*path/newsletter_signup", to: "newsletter_signups#create", via: :put
 
+  resources :sponsors, only: [:index]
+  resources :series, only: [:show]
+  resources :team_members, path: "team", only: [:index, :show]
```
Comment (ColinW520) [inline config/routes.rb:38]: Did you mean to move these?

File: app/services/create_newsletter_signup.rb [hunk 2]
```diff
@@ -2,23 +2,31 @@
...
-    signup_api_call
+    call = sendgrid_api_call
+    if call
+      'success'
+    else
+      'error'
+    end
+  rescue => e
+    'error'
   end
```
Comment (ColinW520) [inline app/services/create_newsletter_signup.rb:25]: Calling `perform` on this service returns just a string... what if this were where you return a result object?

File: config/routes.rb (hunk 1 above)
Comment (willblake01) [reply config/routes.rb:38]: I realized I added newsletter_signup out of alphabetical order and figured this PR would be a good place to move it back into position.

File: spec/services/create_newsletter_signup_spec.rb [hunk 3]
```diff
@@ -13,29 +13,39 @@
...
+            'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
+            'Authorization' => "Bearer " + subject.api_key,
+            'Content-Type'=>'application/json',
+            'User-Agent'=>'sendgrid/5.3.0;ruby'
           }).
         to_return(status: 200, body: "", headers: {})
+
+      subject.perform
+
+      expect(stub).to have_been_requested
```
Comment (ColinW520) [inline spec/services/create_newsletter_signup_spec.rb:48]: 👍

File: app/services/create_newsletter_signup.rb [hunk 4]
```diff
@@ -29,5 +37,20 @@ def signup_api_call
     }
     sg = SendGrid::API.new(api_key: api_key)
     response = sg.client.marketing.contacts.put(request_body: api_call_data)
+    if response.status_code == '202'
+      ServiceResponse.new(
```
Comment (ColinW520) [inline app/services/create_newsletter_signup.rb:41]: Yes. ServiceResponse 🙌

**Review by ColinW520 — APPROVED** [review]

**Pull request merged**